tests/test_api.py

```diff
@@ -194,6 +197,10 @@ def test_reporter(self):
     def test_reporter_with_branches(self):
         sh.coverage('run', '--branch', 'runtests.py')
         results = self.cover.get_coverage()
+
+        if self.oldCoverage and 'branches' not in results:
```

Comment:
Could we make this `if self.oldCoverage` and <TEST_FOR_PY36OR_GREATER>` instead?